### PR TITLE
autocreate firewall rules if HTTP or HTTPS option checked

### PR DESCRIPTION
**Pull request by kenskyfish**

### CODE REVIEW

**Comment by kenskyfish** [comment]
Please review carefully, I'm an amateur. Especially if there's a way to detect firewalls present and then skip instead of just catching error and doing nothing, that'd be nice.

File: app.js [hunk 1]
```diff
@@ -101,6 +101,45 @@ async function createInstance(action, settings) {
...
+            allowed: [{
+                IPProtocol: 'tcp',
+                ports: [80]
+            }]
+        });    
+
+        try {
+            const fwHttpResult = await computeClient.createFirewallRule(firewallResource, waitForOperation);
+            } catch (error) {
+            // throw error;
```
Comment (dtojikulov) [inline app.js:127]: please uncomment this, I have added a check in createFirewallRule method, if firewall rule with the same name is present it throws an error telling that.

File: app.js [hunk 2]
```diff
@@ -101,6 +101,45 @@ async function createInstance(action, settings) {
...
+        firewallResource["targetTags"] = ["https-server"];
+        firewallResource["allowed"] = [{
+            IPProtocol: 'tcp',
+            ports: [443]
+        }];
+    
+        try {
+            const fwHttpsResult = await computeClient.createFirewallRule(firewallResource, waitForOperation);
+            } catch (error) {
+            // throw error;
```
Comment (dtojikulov) [inline app.js:140]: same here, needs to be uncommented.

**Review by dtojikulov — CHANGES_REQUESTED** [review]
Just uncomment the throws in trycatch blocks. I have added a check that firstly determines if the rule with the same name exists and then goes on.

**Pull request closed without merging**